### PR TITLE
fix MPP-566: add docs/fx-integration.md

### PR DESCRIPTION
**Pull request by groovecoder**
This PR fixes #MPP-566.

How to test:
* Read the doc
   * [x] It makes sense.

- [x] ~l10n changes have been submitted to the l10n repository, if any.~
- [x] ~I've added a unit test to test for potential regressions of this bug.~
- [x] I've added or updated relevant docs in the docs/ directory.
- [x] ~All UI revisions follow the [coding standards](https://github.com/mozilla/fx-private-relay/blob/main/docs/coding-standards.md), and use Protocol tokens where applicable (see `/frontend/src/styles/tokens.scss`).~
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).

### CODE REVIEW

File: docs/fx-integration.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+1. Firefox users sign into their browser with their FXA.
+2. Firefox requests an FXA access token with Relay scope for the user.
+3. Firefox sends HTTPS requests to the Relay REST API with the access token.
+
+## Firefox users sign into their browsers with their FXA
+
+Relay has to forward emails to someone's existing email address. So, to use Relay, users
+create a [Firefox Account][sumo-fxa], which requires and verifies an existing email address.
+
+TODO: Link to tech doc for Firefox/FXA integration.
```
Comment (say-yawn) [inline docs/fx-integration.md:17]: (suggestion, blocking) complete this TODO.

File: docs/fx-integration.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+## Firefox requests an FXA access token with Relay scope for the user
+
+Firefox uses [`FxAccounts.getOAuthToken`][fxa-getOAuthToken], which `POST`s to [the FXA
+`/oauth/token` endpoint][fxa-oauth-token] to get an access token with these scopes:
+
+- `profile`
+- `https://identity.mozilla.com/apps/relay`
```
Comment (say-yawn) [inline docs/fx-integration.md:25]:
```suggestion
- `profile`: allows Relay to access the user's profile data which contains the user's primary email, profile pic, and more. See more information about the `profile` scope at [FxA doc](https://mozilla.github.io/ecosystem-platform/reference/oauth-details#profile-data)
- `https://identity.mozilla.com/apps/relay`: enforces the token to be only used by Relay as a [resource server](https://www.oauth.com/oauth2-servers/the-resource-server/)
   - Read more about resource server scoped access key on [FxA's development](https://mozilla.github.io/ecosystem-platform/relying-parties/tutorials/integration-with-fxa#development) doc part 8.
```

File: docs/fx-integration.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+## Firefox sends HTTPS requests to the Relay REST API with the access token
+
+Firefox sends the FXA access token to the Relay REST API as described in
+[the FXA OAuth Token Authentication section of the Relay API auth doc][relay-api-doc-auth].
+
+Firefox uses [Relay's REST API][relay-rest-api]. In particular, it uses these endpoints:
+
+- `POST /api/v1/terms-accepted-user/` when the user opts into Relay integration.
+- `GET /api/v1/profiles/` to get the user's Relay profile data.
+- `GET|POST /api/v1/relayaddresses/` to get and make relay addresses for the user.
```
Comment (say-yawn) [inline docs/fx-integration.md:39]: This is helpful! 👍

**Review by say-yawn — CHANGES_REQUESTED** [review]

File: docs/fx-integration.md (hunk 1 above)
Comment (groovecoder) [reply docs/fx-integration.md:17]: I asked but there isn't a doc for this yet. 😢

**Review by say-yawn — APPROVED** [review]
Helpful documentation on the Firefox Integration work, great job @groovecoder!

**Pull request merged**